src/EnergyPlus/WindowAC.cc

```diff
@@ -372,96 +370,71 @@ namespace WindowAC {
                 }
             }
 
-            state.dataWindowAC->WindAC(WindACNum).FanType = Alphas(7);
             state.dataWindowAC->WindAC(WindACNum).FanName = Alphas(8);
 
-            FanErrFlag = false;
-            ValidateComponent(
-                state, state.dataWindowAC->WindAC(WindACNum).FanType, state.dataWindowAC->WindAC(WindACNum).FanName, FanErrFlag, CurrentModuleObject);
-            if (FanErrFlag) {
-                ShowContinueError(state, format("specified in {} = \"{}\".", CurrentModuleObject, state.dataWindowAC->WindAC(WindACNum).Name));
-                ErrorsFound = true;
-            } else {
-                if (Util::SameString(state.dataWindowAC->WindAC(WindACNum).FanType, "Fan:SystemModel")) {
-                    state.dataWindowAC->WindAC(WindACNum).FanType_Num = DataHVACGlobals::FanType_SystemModelObject;
-                    state.dataHVACFan->fanObjs.emplace_back(
-                        new HVACFan::FanSystem(state, state.dataWindowAC->WindAC(WindACNum).FanName)); // call constructor
-                    state.dataWindowAC->WindAC(WindACNum).FanIndex =
+            state.dataWindowAC->WindAC(WindACNum).fanType =
+                static_cast<DataHVACGlobals::FanType>(getEnumValue(DataHVACGlobals::fanTypeNamesUC, Alphas(7)));
+            if (state.dataWindowAC->WindAC(WindACNum).fanType == DataHVACGlobals::FanType::Invalid) {
+                ShowSevereInvalidKey(state, eoh, cAlphaFields(7), Alphas(7));
```

Comment:
The way this new function (#10066) is used, after converting a choice field to an enum, I don't see how it can ever be called because the choice field should always return a valid enum and therefore not be Invalid. The IP would flag any invalid inputs.

Reply:
I'm changing all of these to asserts. Thx.

Reply:
The assert would never execute. It would be dead code.

Reply:
You're right. Why can't I remember this? FWIW they don't even compile in release code. So they are really a form of documentation more than anything else, but I will delete them.

Reply:
Actually I'm going to leave them in. They make me feel better and they don't affect performance.

Reply:
constexpr int getEnumValue(const gsl::span<const std::string_view> sList, const std::string_view s, const bool b)
    {
        for (unsigned int i = 0; i < sList.size(); ++i) {
            if (sList[i] == s) return i;
        }
        if (b) assert(false);
        return -1;
    }

Reply:
Don't even need the `b` parameter.

Reply:
I agree, if all enums are for choice fields.